server/src/main/java/org/opensearch/identity/tokens/OnBehalfOfClaims.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.tokens;
+
+/**
+ * This class represents the claims of an OnBehalfOf token.
+ */
```

Comment:
Nit: `Add @opensearch.experimental` annotation. Eventually this would be `@opensearch.api` since it would need to be read within the security plugin. 

This PR looks good to me! 🚢